spec/services/creators/vehicle_creation_service_spec.rb

```diff
@@ -4,7 +4,8 @@ module Creators
   RSpec.describe VehicleCreator do
     let(:assessment) { create :assessment, :with_capital_summary }
     let(:capital_summary) { assessment.capital_summary }
-    let(:vehicles_params) { { vehicles: attributes_for_list(:vehicle, 2) }.to_json }
+    let(:vehicles) { attributes_for_list(:vehicle, 2) }
+    let(:vehicles_params) { { vehicles: vehicles.map { |v| v.tap { |h| h[:date_of_purchase] = h[:date_of_purchase].to_s } } } }
```

Comment:
I wish there was a cleaner way of dealing with all these date objects/strings. But I don't think there is. So I guess there's nothing else really to say.

Reply:
There probably is. The challenge is that these functions take JSON as input (which I have converted into Hashes) but they still all want to validate a JSON schema for their inputs. The pattern is very repetitive, I'm sure there's a better way